pvlib/bifacial/infinite_sheds.py

```diff
@@ -0,0 +1,805 @@
+        Ground-reflected irradiance that would reach the row surface if the
+        full ground was visible. poa_gnd_sky accounts for limited view of the
+        sky from the ground. [W/m^2]
+    f_gnd_pv_shade : numeric
+        fraction of ground visible from shaded part of PV surface. [unitless]
+    f_gnd_pv_noshade : numeric
+        fraction of ground visible from unshaded part of PV surface. [unitless]
+
+    Returns
+    -------
+    numeric
+        Ground diffuse irradiance on the row plane. [W/m^2]
+    """
+    return poa_gnd_sky * (f_x * f_gnd_pv_shade + (1 - f_x) * f_gnd_pv_noshade)
+
+
+def _shaded_fraction(solar_zenith, solar_azimuth, surface_tilt,
+                     surface_azimuth, gcr):
+    """
+    Calculate fraction (from the bottom) of row slant height that is shaded
+    from direct irradiance by the row in front toward the sun.
+
+    See [1], Eq. 14 and also [2], Eq. 32.
+
+    .. math::
+        F_x = \\max \\left( 0, \\min \\left(\\frac{\\text{GCR} \\cos \\theta
+        + \\left( \\text{GCR} \\sin \\theta - \\tan \\beta_{c} \\right)
+        \\tan Z - 1}
+        {\\text{GCR} \\left( \\cos \\theta + \\sin \\theta \\tan Z \\right)},
+        1 \\right) \\right)
+
+    Parameters
+    ----------
+    solar_zenith : numeric
+        Apparent solar zenith. [degrees]
+    solar_azimuth : numeric
+        Solar azimuth. [degrees]
+    surface_tilt : numeric
+        Row tilt from horizontal, e.g. surface facing up = 0, surface facing
+        horizon = 90. [degrees]
+    surface_azimuth : numeric
+        Azimuth angle of the row surface. North=0, East=90, South=180,
+        West=270. [degrees]
+    gcr : numeric
+        Ground coverage ratio, which is the ratio of row slant length to row
+        spacing (pitch). [unitless]
+
+    Returns
+    -------
+    f_x : numeric
+        Fraction of row slant height from the bottom that is shaded from
+        direct irradiance.
+
+    References
+    ----------
+    .. [1] Mikofksi, M., Darawali, R., Hamer, M., Neubert, A., and Newmiller,
+       J. "Bifacial Performance Modeling in Large Arrays". 2019 IEEE 46th
+       Photovoltaic Specialists Conference (PVSC), 2019, pp. 1282-1287.
+       doi: 10.1109/PVSC40753.2019.8980572.
+    .. [2] Kevin Anderson and Mark Mikofski, "Slope-Aware Backtracking for
+       Single-Axis Trackers", Technical Report NREL/TP-5K00-76626, July 2020.
+       https://www.nrel.gov/docs/fy20osti/76626.pdf
+    """
+    tan_phi = utils._solar_projection_tangent(
+        solar_zenith, solar_azimuth, surface_azimuth)
+    # length of shadow behind a row as a fraction of pitch
+    x = gcr * (sind(surface_tilt) * tan_phi + cosd(surface_tilt))
+    f_x = 1 - 1. / x
+    # shadow is long enough to fall on row surface if x > 1
+    f_x = np.where(x > 1., f_x, 0.)
+    # when tan_phi < 0, then either sun is behind the array or below the
+    # horizon. Shaded fraction is assigned to be 1 in these cases
+    f_x = np.where(tan_phi >= 0, f_x, 1.)
```

Comment:
This doesn't seem right to me -- how can a criterion that doesn't consider `surface_tilt` determine whether the sun is behind the array?  For `surface_azimuth=180`, `tan_phi<0` might just mean the sun is in the northern sky, which doesn't necessarily imply it is behind the array (consider `surface_tilt=0`). 

Might be easiest to use `(pvlib.irradiance.aoi(...) < 90) & (solar_zenith < 90)` here instead.